postgres/pgxv4/postgres.go

```diff
@@ -54,15 +58,32 @@ type pgDriver struct {
 //
 // "host=my-project:us-central1:my-db-instance user=myuser password=mypass"
 func (p *pgDriver) Open(name string) (driver.Conn, error) {
-	config, err := pgx.ParseConfig(name)
+	dbURI, err := p.getDbURI(name)
 	if err != nil {
 		return nil, err
 	}
-	instConnName := config.Config.Host // Extract instance connection name
-	config.Config.Host = "localhost"   // Replace it with a default value
-	config.DialFunc = func(ctx context.Context, _, _ string) (net.Conn, error) {
-		return p.d.Dial(ctx, instConnName)
-	}
-	dbURI := stdlib.RegisterConnConfig(config)
 	return stdlib.GetDefaultDriver().Open(dbURI)
 }
+
+//getDbURI returns and caches database URI for the connection name
+func (p *pgDriver) getDbURI(name string) (string, error) {
+	var dbURI string
+	var ok bool
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	if dbURI, ok = p.dbURIs[name]; !ok {
```

Comment:
Let's invert the use ok here like so:

``` golang
if dbURI, ok = p.dbURIs[name]; ok {
  return stdlib.GetDefaultDriver().Open(dbURI)
}
```

Then the remainder of the method can be about registering a conn config and finally returning a net.Conn.